bittensor/config.py

```diff
@@ -215,7 +219,7 @@ def __init__(
                             cmd_parser._defaults.clear()  # Needed for quirk of argparse
 
         ## Reparse the args, but this time with the defaults as argparse.SUPPRESS
-        params_no_defaults = config.__parse_args__(
+        params_no_defaults = self.__parse_args__(
             args=args, parser=parser_no_defaults, strict=strict
         )
 
```

Comment:
Just below here where they build the is set dict is wild and is where I'd do some surgery. Rather than diffing 2 dictionaries I'd just build the is_set dict by taking the config keys prior to applying defaults. In other words, build the entire config sans defaults from params_no_defaults then build a dict using params and merge them together at a later step to apply the defaults before passing things down.